pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -30,6 +30,9 @@ type ConfigGeneratorImpl struct {
 	// Must be rebuilt for each push epoch
 	PrecomputedOutboundClustersForGateways map[string][]*xdsapi.Cluster
 	// TODO: add others in future
+
+	// PerRouteFilterConfig has plugin Router Filter config
+	PerRouteFilterConfig map[plugin.Name]interface{}
```

Comment:
See how the XDSOutboundClusters is added to the SidecarScope object. I suggest following the same pattern and caching the mixer routes in each sidecar scope. Thats a safer approach compared to this. 
Ignore the gateway for the moment. That will be undergoing some changes and will become similar to the sidecar scope. The majority of the perf hit is from the sidecars. So even if you cache the sidecar's scope, its a big improvement.

Reply:
OK, added into SidecarScope object now and ignores gateway as you suggested.